src/sickle.ts

```diff
@@ -197,7 +197,7 @@ class Annotator {
         let typeAssertion = <ts.TypeAssertion>node;
         this.maybeEmitJSDocType(typeAssertion.type);
         this.emit('(');
-        this.visit(typeAssertion.expression);
+        this.writeNode(node);
```

Comment:
A random check for similar patterns in other branches shows that we have - 'this.visit(fnDecl.body);' should that be also  this.writeNode(fnDecl.body);?